src/internal/internal_gerbt.cc

```diff
@@ -0,0 +1,470 @@
+                {
+                    scalar_t dummy;
+
+                    A11.tileGetForWriting(ii, jj, LayoutConvert::None);
+                    U1.tileGetForReading(ii, 0, LayoutConvert::None);
+                    V1.tileGetForReading(jj, 0, LayoutConvert::None);
+                    Tile<scalar_t> a11 = A11(ii, jj);
+                    Tile<scalar_t> u1 = U1(ii, 0);
+                    Tile<scalar_t> v1 = V1(jj, 0);
+
+                    Tile<scalar_t> a12;
+                    Tile<scalar_t> a21;
+                    Tile<scalar_t> a22;
+                    Tile<scalar_t> u2;
+                    Tile<scalar_t> v2;
+
+                    if (ii < mt) {
+                        A21.tileGetForWriting(ii, jj, LayoutConvert::None);
+                        U2.tileGetForWriting(ii, 0, LayoutConvert::None);
+                        a21 = A21(ii, jj);
+                        u2 = U2(ii, 0);
+                    }
+                    else {
+                        a21 = Tile<scalar_t>(0, a11.nb(), &dummy, 0, HostNum,
+                                             TileKind::UserOwned, Layout::ColMajor);
+                        u2  = Tile<scalar_t>(0, u1.nb(), &dummy, 0, HostNum,
+                                             TileKind::UserOwned, Layout::ColMajor);
+                    }
+
+                    if (jj < nt) {
+                        A12.tileGetForWriting(ii, jj, LayoutConvert::None);
+                        V2.tileGetForWriting(jj, 0, LayoutConvert::None);
+                        a12 = A12(ii, jj);
+                        v2 = V2(jj, 0);
+                    }
+                    else {
+                        a12 = Tile<scalar_t>(a11.mb(), 0, &dummy, a11.mb(), HostNum,
+                                             TileKind::UserOwned, Layout::ColMajor);
+                        v2  = Tile<scalar_t>(0, v1.nb(), &dummy, 0, HostNum,
+                                             TileKind::UserOwned, Layout::ColMajor);
+                    }
+
+                    if (ii < mt && jj < nt) {
+                        A22.tileGetForWriting(ii, jj, LayoutConvert::None);
+                        a22 = A22(ii, jj);
+                    }
+                    else {
+                        a22 = Tile<scalar_t>(a21.mb(), a12.nb(), &dummy, a21.mb(), HostNum,
+                                             TileKind::UserOwned, Layout::ColMajor);
+                    }
+                    gerbt( a11, a12, a21, a22, u1, u2, v1, v2 );
+                }
+            }
+        }
+    }
+
+    for (int64_t ii = 0; ii < mt_full; ++ii) {
+        for (int64_t jj = 0; jj < nt_full; ++jj) {
+            MPI_Request r;
+            const int64_t tag = 4*(ii*nt_full + jj);
+            if (A11.tileIsLocal(ii, jj)) {
+                // Use undefered task as a selective barrier
+                // TODO change to taskwait depend(...) once OpenMP 5.0 supported
+                #pragma omp task if(false) depend(in:task[ii*nt_full + jj])
```

Comment:
I am not sure I understand why you have an empty task here. Let's discuss it when we review it together.

Reply:
I want to wait for some tasks to complete but not all.  Effectively, it's `#pragma omp taskwait depend(in:task[ii*nt_full + jj])`, but the compiler support for OpenMP 5.0 is pretty bad right now.